runtime/src/main/java/org/corfudb/util/LogicalTicker.java

```diff
@@ -0,0 +1,49 @@
+package org.corfudb.util;
+
+import com.github.benmanes.caffeine.cache.Expiry;
+import com.github.benmanes.caffeine.cache.Ticker;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.corfudb.protocols.wireprotocol.ILogData;
+
+/**
+ * A Ticker subclass for manipulating Caffeine cache expiry
+ * policy using logical time.
+ */
+
+public class LogicalTicker implements Ticker {
+    AtomicLong time;
+
+    public LogicalTicker(AtomicLong time) {
+        this.time = time;
+    }
+
+    @Override
+    public long read() {
+        return time.get();
+    }
+
+    /** Return an expiry policy based on logical time. */
+    public static Expiry<Long, ILogData> caffeineLogicalExpiryPolicy() {
+        // Use a policy of logical time, where the logical clock advances
+        // each time that we witness a TrimmedException.  The "time" for
+        // a cache entry's expiry is its log address.
+
+        return new Expiry<Long, ILogData>() {
+            public long expireAfterCreate(Long key, ILogData l, long currentTime) {
+                return key;
+            }
+
+            public long expireAfterUpdate(Long key, ILogData l,
+                                          long currentTime, long currentDuration) {
+                return key;
+            }
+
+            public long expireAfterRead(Long key, ILogData l,
+                                        long currentTime, long currentDuration) {
+                return key;
```

Comment:
The return value is the duration until the item expires from the current time. So this extends the lifetime of the entry on every read or write by N. From your description, I think you might be only wanting it to be on creation so that entries expire after the clock advances past the threshold. I _think_ you might have thought of this as a point-in-time instead of duration, but am not familiar enough to know for certain.